### ISSUE TITLE
Feature request: PPA

### ISSUE THREAD

**Issue by png2378** [bboozzoo/mconnect#7]
For many people will be useful to have PPA for easy install mconnect on Ubuntu and flavours (Ubuntu GNOME, elementary OS etc).

**Comment by albertvaka** [comment]
Why do you really need mconnect on Ubuntu? Is KDE Connect not good for you?

**Comment by rodneyrod** [comment]
Not speaking for @png2378 but the issue I have with kdeconnect is that on Ubuntu and Debian it pulls in a lot of KDE core packages by default, I shouldn't have to install KMail to use it.

**Comment by png2378** [comment]
@albertvaka, I use elementary OS Loki and original KDE Connect has a lot of dependencies here. 

I used [this indicator](https://github.com/vikoadi/indicator-kdeconnect) on previously version my OS (Freya) and it worked tolerably, and I resigned with a lot of KDE-packages. But now it works unstable and I would like to have an analog written on Vala, who would not had problems on gtk-desktops.

**Comment by bboozzoo** [comment]
Good idea. Though I'm not really a Debian/Ubuntu user. Perhaps someone wants to jump in and do the packaging. Given that mconnect uses autotools I can imagine that the whole process should be rather simple.

**Comment by albertvaka** [comment]
Recent versions of KDE Connect (0.9+) shouldn't have a lot of dependencies on KDE frameworks :/

**Comment by User65k** [comment]
True, now its 100MB compared to 500 (On my setup). Still...

I would also love a PPA!
Maybe you could join forces with this guy:
https://github.com/Bajoja/indicator-kdeconnect
https://launchpad.net/~varlesh-l/+archive/ubuntu/indicator-kdeconnect/+packages

**Comment by fisadev** [comment]
+1 to this. I don't want all the KDE services running in my system just to be able to connect with my phone. mconnect seems like a far better option, if I was able to install it without having to learn how to compile it.

**Comment by RicardoEPRodrigues** [comment]
If not a PPA then a Snap or Flatpak. :smile:

**Comment by bboozzoo** [comment]
Snap and Flatpak make more sense, i"ll try to look into that. I suppose it should be possible to cover both.

**Comment by bboozzoo** [comment]
I started with snaps first. Relevant changes are in #31. So far I have somewhat mixed feelings. The resulting snap is very large, even though  have tried to use a shared part delivering gnome-3.26 libraries (or that's what I understood).  See this:
```
 65M -rw-r--r-- 1 maciek maciek  65M 10-29 19:21 m-connect_0.3+git_amd64.snap
```

Also, the name `mconnect` is already registered, so I had to register the snap under `m-connect` name.

Building the snaps turned out to a bit annoying as well. You actually need Ubuntu for snapcraft to work. I've setup 17.10 under LXC, but ran into problems with built snaps. Apparently, the build process is not using `core` as one would expected (or I missed some command line switch) but the local system. So the binaries were linked against libraries in 17.10, while `core` seems to be 16.04. Obviously `snap run m-connect.mconnect` only raised linker issues (as expected).

OTOH, once I got it to build and run, I was able to connect with KDE connect running on a phone. Controlling the daemon through `mconnectctl` worked too.

**Comment by fisadev** [comment]
@bboozzoo if I can help testing it in any way, just tell me. My current setup is Ubuntu 16.10, and I'll probably switch to 17.10 soon.

**Comment by balsoft** [comment]
I am currently working on a Nix package. Most of the stuff seems to work, I just can't get AT-SPI to work properly. Also, would be nice if someone could help out by testing this outside NixOS.
